dispatcher/brokers/pg_notify.py

```diff
@@ -25,6 +26,29 @@ def create_connection(**config) -> psycopg.Connection:
     return psycopg.Connection.connect(**config)
 
 
+def current_notifies(conn: psycopg.Connection) -> Generator[psycopg.connection.Notify, None, None]:
+    """Altered version of .notifies method from psycopg library
+
+    Taken from AWX, only used for synchronous listening, same notify-or-timeout problem
+    This removes the outer while True loop so that we only process
+    queued notifications
+    """
+    with conn.lock:
+        try:
+            ns = conn.wait(psycopg.generators.notifies(conn.pgconn))
+        except psycopg.errors._NO_TRACEBACK as ex:
+            raise ex.with_traceback(None)
+    for pgn in ns:
+        if hasattr(conn.pgconn, '_encoding'):
+            # later, like 3.2+ versions
+            enc = conn.pgconn._encoding
+        else:
+            # earlier versions
+            enc = psycopg._encodings.pgconn_encoding(conn.pgconn)
```

Comment:
This generalizes relatively recent work from https://github.com/ansible/awx/pull/15705